app/src/main/java/com/hedvig/app/authenticate/insurely/InsurelyAuthViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.hedvig.app.authenticate.insurely
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class InsurelyAuthViewModel(
+    private val reference: String,
+    private val getDataCollectionUseCase: GetDataCollectionUseCase
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(ViewState())
+    val viewState: StateFlow<ViewState>
+        get() = _viewState
+
+    data class ViewState(
+        val autoStartToken: String? = null,
+        val authStatus: DataCollectionResult.Success.CollectionStatus? = null,
+        val error: DataCollectionResult.Error? = null,
+    )
```

Comment:
Does the Error case maybe need to be mutually exclusive from the other states?
Is there a particular reason an error can exist at the same time that either of the other two variables?
If not a sealed class maybe fits this case better, what do you think?